ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockHeader.java

```diff
@@ -142,33 +144,50 @@ public void writeTo(final RLPOutput out) {
     out.writeBytes(extraData);
     out.writeBytes(mixHash);
     out.writeLong(nonce);
-
+    if (baseFee != null) {
+      out.writeLongScalar(baseFee);
+    }
     out.endList();
   }
 
   public static BlockHeader readFrom(
       final RLPInput input, final BlockHeaderFunctions blockHeaderFunctions) {
     input.enterList();
-    final BlockHeader blockHeader =
-        new BlockHeader(
-            Hash.wrap(input.readBytes32()),
-            Hash.wrap(input.readBytes32()),
-            Address.readFrom(input),
-            Hash.wrap(input.readBytes32()),
-            Hash.wrap(input.readBytes32()),
-            Hash.wrap(input.readBytes32()),
-            LogsBloomFilter.readFrom(input),
-            Difficulty.of(input.readUInt256Scalar()),
-            input.readLongScalar(),
-            input.readLongScalar(),
-            input.readLongScalar(),
-            input.readLongScalar(),
-            input.readBytes(),
-            Hash.wrap(input.readBytes32()),
-            input.readLong(),
-            blockHeaderFunctions);
+    final Hash parentHash = Hash.wrap(input.readBytes32());
+    final Hash ommersHash = Hash.wrap(input.readBytes32());
+    final Address coinbase = Address.readFrom(input);
+    final Hash stateRoot = Hash.wrap(input.readBytes32());
+    final Hash transactionsRoot = Hash.wrap(input.readBytes32());
+    final Hash receiptsRoot = Hash.wrap(input.readBytes32());
+    final LogsBloomFilter logsBloom = LogsBloomFilter.readFrom(input);
+    final Difficulty difficulty = Difficulty.of(input.readUInt256Scalar());
+    final long number = input.readLongScalar();
+    final long gasLimit = input.readLongScalar();
+    final long gasUsed = input.readLongScalar();
+    final long timestamp = input.readLongScalar();
+    final Bytes extraData = input.readBytes();
+    final Hash mixHash = Hash.wrap(input.readBytes32());
+    final long nonce = input.readLong();
+    final Long baseFee = input.isEndOfCurrentList() ? null : input.readLongScalar();
```

Comment:
This too should be feature flag guarded.  So it it isn't enabled we will get RLP errors if they try to use it w/o the flag.

Reply:
Sure. I will do that when step 0 gets merged.

Reply:
Done in https://github.com/hyperledger/besu/pull/619/commits/7d7f59b2d35fcf6fb4ec6853a732de388a52859e